glean-core/src/metrics/remote_settings_config.rs

```diff
@@ -16,25 +16,29 @@ use serde::{Deserialize, Serialize};
 /// }
 /// ```
 #[derive(Serialize, Deserialize, Debug, Clone, Default)]
-pub struct MetricsEnabledConfig {
+pub struct RemoteSettingsConfig {
     /// This is a `HashMap` consisting of base_identifiers as keys
     /// and bool values representing an override for the `disabled`
     /// property of the metric, only inverted to reduce confusion.
     /// If a particular metric has a value of `true` here, it means
     /// the default of the metric will be overriden and set to the
     /// enabled state.
-    #[serde(flatten)]
-    pub metrics_enabled: HashMap<String, bool>,
+    pub metrics_enabled: Option<HashMap<String, bool>>,
+
+    /// This is a `HashMap` consisting of ping names as keys and
+    /// boolean values representing on override for the default
+    /// enabled state of the ping of the same name.
+    pub pings_enabled: Option<HashMap<String, bool>>,
```

Comment:
If I remember my `std::collections` rules of thumb correctly, in most cases we're to not bother with `Option<ACollectionType>` and instead use emptiness (since empty collections do not allocate). As a bonus it ought to make the metrics setting code quite a lot simpler.

Reply:
Then I would need an intermediate representation to assume the serde deserialization. Here Optional is also letting serde know that this may or may not be present in the configuration that is passed through the API.

Reply:
Can you go into more detail? I assume `#[serde(default)]` solves for configurations that are missing `pings_enabled` or `metrics_enabled`?

Reply:
ref. https://serde.rs/field-attrs.html#default